app/assets/javascripts/components/slice_buttons.js

```diff
@@ -0,0 +1,112 @@
+(function() {
+  window.shared || (window.shared = {});
+  var dom = window.shared.ReactHelpers.dom;
+  var styles = window.shared.styles;
+  var colors = window.shared.colors;
+
+  var SliceButtons = window.shared.SliceButtons = React.createClass({
+    displayName: 'SliceButtons',
+      
+    propTypes: {
+      students: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
+      filters: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
+      filtersHash: React.PropTypes.func.isRequired,
+      activeFiltersIdentifier: React.PropTypes.func.isRequired,
```

Comment:
I think `SliceButtons` only needs the values for `filtersHash` and `activeFiltersIdentifier`.  So you can have the call sites evaluate those functions and then pass the plain value into `SliceButtons` (the same way that `students` works).

Like this:
```
createEl(SliceButtons, {
  students: this.filteredStudents(),
  filters: this.state.filters,
  filtersHash: this.filtersHash(),
  activeFiltersIdentifier: this.activeFiltersIdentifier(),
  onClearFilters: this.onClearFilters
})
```